### DISCUSSION TITLE
Prerender Error debugging help

### DISCUSSION THREAD

**Discussion by NotMichaelsGitHub** [vercel/next.js#44171] [Help]
Hey guys, I'm trying to update a 2 year old project, and I've been having so much trouble after updating the dependencies 

Right now after running 'next build', I keeps seeing:
 `Error occurred prerendering page "/japan-vlogs". Read more: https://nextjs.org/docs/messages/prerender-error    
TypeError: Cannot read properties of null (reading 'useMemo')
`
for every page in my project 

How do I even begin to debug this? The docs site has 5 different options and none of which say anything about the react dom server browser or use memo

![image](https://user-images.githubusercontent.com/14267204/208556034-37960a2e-1039-412e-93a1-53672519ded4.png)
![image](https://user-images.githubusercontent.com/14267204/208556050-a6adf85c-75fe-4d6a-b3f4-73dc306fa1e8.png)
![image](https://user-images.githubusercontent.com/14267204/208556080-38cce618-3f6d-4634-a89b-a450fcc40c68.png)
![image](https://user-images.githubusercontent.com/14267204/208556119-39defdbb-f361-4358-a5f5-63d4aaeb4cb4.png)

**Comment by ElliotGibson** [comment]
The error is occuring with your redux provider by the looks of things - it should be in _app (I'm assuming it is) - otherwise the modules look fine, I'd probably try and do a clean install and see if the problem persists

The pre-render error is just noting where/when the error occured, it's odd that it didn't throw when building the client bundle...

It may we be worth adding _app to the issue :)

**Comment by icyJoseph** [comment]
Did you get around fixing this? I tried your repo ant it builds without issue. I'd say thought, that you should remove the `.next` folder, and add it to the `.gitignore` file.

**Comment by lundjrl** [comment]
@NotMichaelsGitHub I'm running into this issue as well. Prerendering pages at build-time errors out with:
`TypeError: Cannot read properties of null (reading 'useMemo')
    at exports.useMemo (ROOT/node_modules/next/dist/compiled/react/cjs/react.production.min.js:29:208)
    at Router (ROOT/.next/server/chunks/936.js:56292:38)`

Running:
- "next": "^13.1.6"
- "react": "^18.2.0"
- "react-dom": "^18.2.0",

Granted, I am using NX for monorepo things. But running `npx nx build <project-name> --dev` fails on prerender but a production build succeeds. 

I've attempted removing the .next folder and dist from nx. 

I'm not using useMemo anywhere in my project. 

Thank you in advance for any traction this may generate.

EDIT:
Running the typical Next commands does prerender pages correctly. NX seems to be the issue here. If I find an issue or discussion, I can add it to this comment in case anyone else is running into this. 

EDIT 2:
If using a mono-repo setup, make sure that there's not an outstanding node_modules folder in your next.js project. I have a setting in my vscode settings.json file which hides this folder.
